### ISSUE TITLE
Browser detection takes 10 seconds on every Cypress open launch

### ISSUE THREAD

**Issue by bahmutov** [cypress-io/cypress#17077]
Brand new top MacBook Pro (Intel)

When doing `npx cypress open` every launch spins for about 10 seconds while the project loads, and I think this is all because of the browser detection. Confirmed with `DEBUG=cypress:launcher,cypress:server:browsers npx cypress open` logs in the movie below

https://user-images.githubusercontent.com/2212006/123111988-59b9a700-d40b-11eb-86e4-6bfd2c8ec9c3.mp4

Typical log fragment for one browser

```
  cypress:launcher checking one browser edge +1ms
  cypress:launcher looking up edge on darwin platform +0ms
  cypress:launcher looking for app Contents/MacOS/Microsoft Edge id com.microsoft.Edge +0ms
  cypress:launcher looking for bundle id com.microsoft.Edge using command: mdfind 'kMDItemCFBundleIdentifier=="com.microsoft.Edge"' | head -1 +0ms
  cypress:launcher checking one browser edge +345ms
  cypress:launcher looking up edge on darwin platform +0ms
  cypress:launcher looking for app Contents/MacOS/Microsoft Edge id com.microsoft.Edge +0ms
  cypress:launcher looking for bundle id com.microsoft.Edge using command: mdfind 'kMDItemCFBundleIdentifier=="com.microsoft.Edge"' | head -1 +0ms
  cypress:launcher could not find com.microsoft.Edge +354ms
  cypress:launcher looking for application /Applications/Microsoft Edge.app +0ms
  cypress:launcher reading property file "/Applications/Microsoft Edge.app/Contents/Info.plist" +0ms
  cypress:launcher could not find com.microsoft.Edge +1ms
  cypress:launcher looking for application /Applications/Microsoft Edge.app +0ms
  cypress:launcher reading property file "/Applications/Microsoft Edge.app/Contents/Info.plist" +0ms
  cypress:launcher could not read Info.plist { pl: '/Applications/Microsoft Edge.app/Contents/Info.plist', e: [Error: ENOENT: no such file or directory, open '/Applications/Microsoft Edge.app/Contents/Info.plist'] { errno: -2, code: 'ENOENT', syscall: 'open', path: '/Applications/Microsoft Edge.app/Contents/Info.plist' } } +0ms
  cypress:launcher could not detect edge using traditional Mac methods +0ms
  cypress:launcher trying linux search +0ms
  cypress:launcher finding version string using command "microsoft-edge --version" +0ms
  cypress:launcher could not read Info.plist { pl: '/Applications/Microsoft Edge.app/Contents/Info.plist', e: [Error: ENOENT: no such file or directory, open '/Applications/Microsoft Edge.app/Contents/Info.plist'] { errno: -2, code: 'ENOENT', syscall: 'open', path: '/Applications/Microsoft Edge.app/Contents/Info.plist' } } +348ms
  cypress:launcher could not detect edge using traditional Mac methods +0ms
  cypress:launcher trying linux search +0ms
  cypress:launcher finding version string using command "edge --version" +0ms
  cypress:launcher Received error detecting browser binary: "microsoft-edge" with error: spawn microsoft-edge ENOENT +355ms
  cypress:launcher Received error detecting browser binary: "edge" with error: spawn edge ENOENT +0ms
  cypress:launcher browser edge not installed +0ms
  cypress:launcher browser edge not installed +0ms
```

- we should probably print a single debug statement "browser detection took Xms"
- we should probably speed this up. Maybe try finding browsers in parallel?

```
Cypress Version: 7.5.0
System Platform: darwin (20.3.0)
System Memory: 34.4 GB free 5.39 GB
```

**Comment by ungb** [comment]
Is there any workaround for this or is the fix to downgrade to 7.4.0?

**Comment by jennifer-shehane** [comment]
This was fixed in Cypress 8.3.1
